doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Note: The reference application supports in a tricky way: the logarithm adaptation is always combined with the affine adaptation)
+(Why: To avoid combination explosion of parameter adaptations.
+      The idea is to builtin basic function and let the user compose them to meet it's need.)
+
+## Identifiers
+Every parameter MUST have a identifier that uniquely identifies it.
+(Why: to identify a parameter outside the framework)
+
+This identifier SHOULD be a string.
+(Why: so that the (human) user can identify a parameter with ease.)
+
+Two PF instances with the same parameters MUST have the same identifier for those parameters.
+Ie. This identifier should be the same across all instances with the same configuration.
+(Why: Persistence of parameter identifier across PF instances with the same configuration.
+      To identify parameters independently of the host machine and PF instance)
+
+### Parameter tree
+A parameter SHOULD be structured in a tree. Each parameter being a distinct tree leaf.
+(Why: Tree is a simple data structure that can be easily represented and is enough to map
+      underlined layers.
+      It is not a MUST because an other structure could be chose like a list or a graph.
+      It would not change the rest of the PF properties thank to the unique identifier requirement.)
+
+Each node of the tree SHOULD have its own identifier with the same
+characteristics (type, independence...) than a parameter.
+(Why: To represent the tree without treating the leaf nodes specifically.)
+
+The identifier of each node of the tree SHOULD be a combination of its parents.
+More specifically, if the identifier is a string it should be formated the same
+way as a file system path. Eg the parameter named "/root/child1/4/parameter1".
+(Why: Usual syntax to address trees.)
+
+
+# Syncer
+
+The PF philosophy is to map the hardware characteristics to parameters.
+
+A syncer MUST be mapped to one or more parameters.
+(Why: The hardware minimal access may be bigger than one parameter.)
+
+One parameter MUST NOT be mapped to two or more syncer.
+(Why: Which syncer should be responsible to retrieve the initial parameter value
+      if they are multiple per parameter?)
+
+A syncer MUST support retrieving the mapped parameters value from the mapped hardware.
+(Why: to retrieve a parameter value at the start of the PF.)
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
```

Comment:
As far as I know there is only one syncer that use this feature in the reference implementation. I added:
> For example a syncer might need each to know each associated parameter type to send it to the hardware.